xdsl/dialects/arith.py

```diff
@@ -230,7 +230,7 @@ class RemUI(SignlessIntegerBinaryOp):
 
 
 @irdl_op_definition
-class RemSI(IntegerBinaryOp):
+class RemSI(SignlessIntegerBinaryOp):
```

Comment:
@PapyChacal Do you remember why this was previously `IntegerBinaryOp`? I just looked at the documentation, and while the operation treats the inputs as signed, lhs and rhs are signless-like. Was this intentional? Do you think this change is safe?